uftrace.c

```diff
@@ -292,21 +292,72 @@ static void parse_debug_domain(char *arg)
 	free(saved_str);
 }
 
+static int get_digits(uint64_t num)
+{
+	int digits = 0;
+
+	do {
+		num /= 10;
+		digits++;
+
+		if (digits == 9)
+			break;
+	} while (num != 0);
+
+	return digits;
+}
+
 static uint64_t parse_time(char *arg)
 {
-	char *unit;
-	uint64_t val = strtoull(arg, &unit, 0);
+	char *unit, *pos;
+	int i, nfore_zero = 0, exp = 0;
+	uint64_t val, decimal = 0;
 
+	pos = strchr(arg, '.');
+	if (pos == NULL) {
+		val = strtoull(arg, &unit, 0);
+	}
+	else {
+		val = strtoull(arg, &pos, 0);
+		while (*(++pos) == '0')
+			nfore_zero++;
```

Comment:
How to know the number of zeroes below decimal point without the above method ?
Is there other way ? using strchr() or strrchr() ?